sdks/python/apache_beam/transforms/util_test.py

```diff
@@ -276,6 +279,13 @@ def process(self, element):
     with self.assertRaisesRegex(ValueError, r'window.*None.*add_timestamps2'):
       pipeline.run()
 
+class AddTimestamp(beam.DoFn):
+  def process(self, element, timestamp=beam.DoFn.TimestampParam):
+    yield beam.window.TimestampedValue(element, timestamp)
```

Comment:
This is a no-op.

Reply:
My understanding is that it should be a no-op, if Reshuffle preserves timestamps. This is what this test is testing. 
Its Java parity is the testReshufflePreservesTimestamps in file
beam/sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ReshuffleTest.java
which wrapped the string element into TimestampedValue twice. The first time the element becomes TimestampedValue<String>; the second time the element becomes 
TimestampedValue<TimestampedValue<String>>.
Python doesn't have nested TimestampedValue type and doesn't have getTimestamp() either, so I used beam.DoFn.TimestampParam to get the timestamp twice, before and after Reshuffle.  Assuming beam.DoFn.TimestampParam is always the current timestamp bounded with an element, then it should work.

Reply:
Yielding a TimestampedValue just sets the timestamp to that value. Essentially, you're getting an element and its timestamp, and returning that same element with its same timestamp. You then follow this with FormatWithTimestamp gets the timestamp and actually does something with it. You can just cut this function and set 

`after_reshuffle = before_reshuffle | beam.Reshuffle()`

Reply:
I agree. Thank you for pointing it out!